### ISSUE TITLE
[Relay support] @argumentDefinitions

### ISSUE THREAD

**Issue by Albert-Gao** [graphql/vscode-graphql#231]
## Steps to Reproduce the Problem Or Description

1. After adding the Relay Definition from this: https://github.com/graphql/vscode-graphql/issues/229#issuecomment-698659570
1. Relay specific directives won't give any problems anymore, for example: `@connection`
1. However, `@argumentDefinitions` has a problem:
![image](https://user-images.githubusercontent.com/18282328/94214308-556c6600-ff2d-11ea-857f-a3c8117041a2.png)
1. As you can see, the `@argumentDefinitions` itself can be identified, but the arguments inside can not be, the definition looks like this: `directive @argumentDefinitions on FRAGMENT_DEFINITION`

I think it might be better if we can ignore validation for certain directives details? For example, we can validate `@argumentDefinitions` but not its specific arguments list, because they are purely client-side spec unless you write it somewhere, not to mention it could change according to where it is being used.

## Specifications

GraphQL for VSCode Extension Version: latest from store
Version: 1.49.1
Commit: 58bb7b2331731bf72587010e943852e13e6fd3cf
Date: 2020-09-16T23:21:17.533Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.6.0

**Comment by acao** [comment]
innnteresting! yes this is gonna be tricky, since we need to be able to validate directive arguments generally. i can’t imagine anything else besides ignore comment/blocks

**Comment by Albert-Gao** [comment]
Lucky me, in my case, currently all my top-level pagination query has the same pattern, so I changed the GraphQL type definition to something like this:


```graphql
type ArgType {
  type: String!
  defaultValue: Int
}

directive @argumentDefinitions(
  orgId: ArgType
  first: ArgType
  after: ArgType
  last: ArgType
  before: ArgType
) on FRAGMENT_DEFINITION
```

No error now
![image](https://user-images.githubusercontent.com/18282328/94337294-82a83980-003d-11eb-9818-a171868705d9.png)

It's not ideal, since the definition is wrong, the type of `defaultValue` should only be Int when `type` is `Int`, but enough to silence the linting. :D 

This is a tricky one.

This extension should be generic, so I suppose we might be able to do something like: in the `graphql.config.js`:

```javascript
module.exports = {
  extensions: {
      skipCheckingDirectives:[ 'argumentDefinitions' ]
  },
}
```

**Comment by acao** [comment]
relay support is coming in a seperate plugin!

**Comment by acao** [comment]
and with that, we shouldn't need to skip any directive validation, as it should work the same as the apollo vscode plugin does? that one validates this directive properly, correct?

**Comment by Albert-Gao** [comment]
not quite sure, I already sunsetted all apollo setup ...

@alloy does the vscode-apollo-relay validate the details of `@argumentionDetail`? I think not? Do not know.

BTW: You playing Horizon Zero Dawn? The main character is named as Alloy :D haha, love that game
![image](https://user-images.githubusercontent.com/18282328/95391423-86985d80-0953-11eb-8b34-271f7eed2f7a.png)

**Comment by alloy** [comment]
Yes, the existing `vscode-apollo-relay` validation rule does a bunch of `@argumentDefinitions` validations! 👍 

As for Horizon Zero Dawn, I do not but it’s great to know that they are making characters after me 😛

**Comment by nibblesnbits** [comment]
Is there any news on this?  I'm loving the extension, but I use Relay and it's always complaining about the relay-specific stuff.

**Comment by acao** [comment]
good question!

we have it mostly ported here:

https://github.com/relay-tools/relay-graphql-js/tree/main/packages/graphql-config

i haven't had a chance to test it myself, i think it still needs some tweaks

**Comment by nibblesnbits** [comment]
How would one go about including this when using the extension?

**Comment by acao** [comment]
@nibblesnbits instructions are [in the readme ](https://github.com/relay-tools/relay-graphql-js/tree/main/packages/graphql-config#usage)!
